### ISSUE TITLE
Storybook not following a symlink for config directory

### ISSUE THREAD

**Issue by stramel** [storybooks/storybook#4083]
If you are reporting a bug or requesting support, start here:
### Bug or support request summary

I am using a monorepo and to try to re-use my storybook config directory I symlinked the directory from the root. However, it does not see it or follow it.

NOTE: I am using lerna w/ yarn workspaces.

### Steps to reproduce

1. Create a top-level working storybook config `.storybook`
1. Make a package in a nested folder (something like `packages/core/image`)
1. run `ln -s .storybook packages/core/image` from the root directory
1. run `npm run storybook` (`start-storybook -p 9009`) from the sub directory.

### Please specify which version of Storybook and optionally any affected addons that you're running

- @storybook/react ^3.4.8

### Affected platforms 

MacOS X High Sierra 10.13.6
Node v8.11.3

### Screenshots / Screencast / Code Snippets (Optional)

```sh
yarn run v1.9.4
$ start-storybook -p 9009 -c .storybook
info @storybook/react v3.4.10
info
web/node_modules/@storybook/react/dist/server/config.js:31
    throw err;
    ^

Error: => Create a storybook config file in ".storybook/config.js".
    at exports.default (web/node_modules/@storybook/react/dist/server/config.js:30:15)
    at exports.default (    at buildDev (web/node_modules/@storybook/core/dist/server/build-dev.js:163:36)
    at Object.<anonymous> (web/node_modules/@storybook/react/dist/server/index.js:23:22)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
    at Module.require (module.js:596:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (web/node_modules/@storybook/react/bin/index.js:3:1)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Comment by stramel** [comment]
Appears to be failing here: https://github.com/storybooks/storybook/blob/562794e65bb4bead1fdfca923df5c2ab41d63de7/lib/core/src/server/config/utils.js#L45

**Comment by igor-dv** [comment]
@stramel,  a long time has passed since you opened this issue, can you confirm this is still happening with v4?

**Comment by stramel** [comment]
I will try to update my stuff to v4 and see if it is still an issue.

**Comment by stramel** [comment]
Sorry for the delay. It does appear to be fixed by storybook 4.0.
